app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoFragment.kt

```diff
@@ -122,23 +118,9 @@ private class PocketVideoAdapter(
         titleView.text = item.title
         PicassoWrapper.client.load(item.thumbnailURL).into(videoThumbnailView)
 
-        @Suppress("TooGenericExceptionCaught") // See below.
-        val itemURI = try {
-            URI(item.url)
-        } catch (e: Exception) { // Apparently Kotlin doesn't have multi-catch.
-            when (e) {
-                is URISyntaxException, is NullPointerException -> null
-                else -> throw e
-            }
-        }
-        domainView.text = if (itemURI == null) {
-            item.url
-        } else {
-            // The first time this method is called ever, it may block until the file is cached on disk.
-            // We pre-cache on startup so I'm hoping this isn't an issue.
-            StrictMode.allowThreadDiskReads().resetAfter {
-                FormattedDomain.format(itemView.context, itemURI, false, 0)
-            }
+        for (x in item.authors.keys()) {
+            val authorObject = JSONObject(item.authors[x].toString())
+            domainView.text = authorObject.get("name").toString()
```

Comment:
(I know we weren't doing some of this before, but since we're changing this code this is a good time to improve it)

I have two problems with this approach.
1) we aren't handling failures (what if the `JSONObject` doesn't have a "name" field?)
2) we are doing JSON parsing in the middle of our app, rather than at the edge

1: Anything that goes across the network should be considered unsafe.  We can't be sure that these objects will come in at all, or that any particular fields won't be null or malformed.  We want to handle these cases wherever we parse this JSON (in this case, handling probably means providing an empty string on failure).

2: As much as possible, we want our domain objects to have exactly what we need, no more and no less.  This isn't always going to match the shape of the JSON (we might drop unnecessary fields, merge two together, add timestamps, etc).  The goal is to be able to just use these without ceremony in the middle of our app(e.g., `domainView.text = video.author`).  It's the parser's job to figure out how to convert from JSON to what we actually need.  In this case, the parser can break out what's in the JSON, figure out which author value to use, and provide an empty string on failure.

One big advantage to this is that it simplifies a lot of how our app functions.  If you handle all the dirtiness of IO on the edge of the application, nothing else has to care about the nullability and errors and so on.  They just use the data.  That prevents us from having to duplicate the same messy logic over and over again, and also makes it really easy to find where problems are happening.  If parsing is failing, check the parser.

TL;DR
- [ ] `FeedItem.Video.author` should be a non-nullable String
- [ ] Getting the author and name fields should be done by the PocketVideoParser
- [ ] We should handle the case where those fields don't exist (probably by passing an empty String)